resources/osl_element.rb

```diff
@@ -0,0 +1,47 @@
+# Element Web Application
+# This resource allows for the deployment of Element, a Matrix client, as a website.
+
+resource_name :osl_element
+provides :osl_element
+unified_mode true
+
+default_action :create
+
+property :domain, String, name_property: true
+property :matrix_domain, String
+
+action :create do
+  include_recipe 'osl-docker'
+  include_recipe 'osl-nginx'
+
+  hostname node['hostname'] do
+    aliases [new_resource.domain]
+  end
+
+  directory '/srv/element'
```

Comment:
I'd prefer to put it under `/opt`
```suggestion
  directory '/opt/element'
```